search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/TypedSearchQueryTransformer.java

```diff
@@ -51,8 +51,8 @@ public SearchQueryRequest toSearchQueryRequest(
 
     final var sort = query.sort();
     final var sorting = toSearchSortOptions(sort, reverse);
-    if (sorting != null && !sorting.isEmpty()) {
-      builder.sort(sorting);
+    if (!sorting.isEmpty()) {
+      builder.sort(query.retainValidSortings(sorting));
```

Comment:
@maryarm the invalid sorting fields are already reported in https://github.com/camunda/camunda/blob/778d3f04b151c083ffde088126a1e94aafc56602/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java#L884-L899
do you expect to have other flow that can bring any invalid sort fields?

Reply:
Maybe I can reuse this method, Although my case is not reporting to the rest caller, the problem is that there is a logic that adds default sorting with the field "key" and as Authorization doesn't have the search is throwing exception